src/tools/kata-ctl/src/types.rs

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) 2022 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use anyhow::{anyhow, Result};
+use strum_macros::Display;
+
+// Builtin check command handler type
+pub type BuiltinCmdFp = fn(args: &str) -> Result<()>;
+
+// Used to show whether a check needs to run with elevated (super-user) privileges, or whether it can run
+#[derive(strum_macros::Display)]
+pub enum PermissionType {
+    Privileged,
+    NonPrivileged,
+}
+
+// Command metadata for the checks
+pub struct CheckItem<'a> {
```

Comment:
Could you add a comment explaining that each `CheckItem` is implicitly architecture-specific: each architecture returns a list of `CheckItem`'s using that architectures implementation of `get_checks()`.